pangeo_forge_recipes/storage.py

```diff
@@ -106,6 +109,42 @@ def from_url(cls, url: str):
         assert len(root_paths) == 1
         return cls(fs, root_paths[0])
 
+    def _get_fsspec_implementation(self) -> str:
+        # can't use `isinstance` checks here b/c deps might not be installed
+        return type(self.fs).__name__
+
+    def get_fsspec_remote_protocol(self):
+        """fsspec implementation-specific remote protocal.
+
+        other implementations can be added here for a general case solution
+        """
+        fs_type_name = self._get_fsspec_implementation()
+        if fs_type_name == "S3FileSystem":
+            return "s3"
+
+    def get_fsspec_credential_kwargs(self):
```

Comment:
a hack: we don't want to manage something gross like this for all implementations